cdap-app-fabric/src/main/java/co/cask/cdap/metadata/DefaultMetadataAdmin.java

```diff
@@ -164,9 +180,9 @@ public void validateTags(Id.NamespacedId entityId, String ... tags) throws Inval
 
   // Validate the key should not be be reserved {@link BusinessMetadataDataset.TAGS_KEY}
   private void validateTagReservedKey(Id.NamespacedId entityId, String key) throws InvalidMetadataException {
-    if (BusinessMetadataDataset.TAGS_KEY.equals(key.toLowerCase())) {
+    if (MetadataDataset.TAGS_KEY.equals(key.toLowerCase())) {
```

Comment:
So these tag keys are like keywords for metadata and we didn't allow user's to set metadata with these key right ? 
Now since we are adding system metadata  and even though they will write to different tables but when a user does getTags() we return a set so does this keys needs to be updated to now allow system ones too ? or are  the keys uniquely identified by with their scope ?